File: ai/examples/logic_error.py
def solution(scores):
    total = 0
    for i in range(len(scores)):
        total += scores[i]
    average = total // len(scores)

    count = 0
    for s in scores:
        if s > average:
            count += 1
    return count

print(solution([70, 80, 90, 60, 75]))


{
  "problem_title": "평균 이상 학생 수",
  "problem_description": "평균 이상인 학생 수 구하기",
  "language": "python",
  "student_code": "def solution(scores):\n    total = 0\n    for i in range(len(scores)):\n        total += scores[i]\n    average = total // len(scores)\n\n    count = 0\n    for s in scores:\n        if s > average:\n            count += 1\n    return count\n\nprint(solution([70, 80, 90, 60, 75]))",
  "test_result": "Wrong Answer",
  "failed_cases": [
    {
      "input": "[70, 80, 90, 60, 75]",
      "expected_output": "3",
      "actual_output": "2",
      "reason": "평균과 같은 값 제외"
    }
  ],
  "judge_message": "논리 오류 발생"
}

File: ai/examples/logic_error.java
{
  "problem_title": "합계 계산",
  "problem_description": "배열 합 구하기",
  "language": "java",
  "student_code": "public class Main {\n    public static void main(String[] args) {\n        int[] arr = {1,2,3};\n        int sum = 0;\n        for(int i=0;i<arr.length-1;i++){\n            sum += arr[i];\n        }\n        System.out.println(sum);\n    }\n}",
  "test_result": "Wrong Answer",
  "failed_cases": [
    {
      "input": "",
      "expected_output": "6",
      "actual_output": "3",
      "reason": "마지막 값 누락"
    }
  ],
  "judge_message": "논리 오류"
}